File: JavaScript/121.js
/**
 * @no 121
 * @name Best Time to Buy and Sell Stock
 */
let maxProfit = (prices) => {
    let ans = 0;

    for (let i = 0 ; i < prices.length; i++) {
        let arr = [];
        for (let j = i + 1; j < prices.length; j++)
            arr.push(prices[j] - prices[i]);

        let max = 0;
        for (let j = 0; j < arr.length; j++)
            if (arr[j] > max) max = arr[j];

        ans = max > ans ? max : ans;
    }

    return ans;
};

File: Java/121.java
/**
 * @no 121
 * @name Best Time to Buy and Sell Stock
 */
class Solution {
    public int maxProfit(int[] prices) {
        if (prices.length == 0) {
            return 0;
        }
        int minPrices = prices[0];
        int result = 0;
        for (int i = 0; i < prices.length; i++) {
            if (prices[i] < minPrices) {
                minPrices = prices[i];
            }
            result = Math.max(result, prices[i] - minPrices);
        }
        return result;
    }
}